src/test/java/uk/gov/companieshouse/itemhandler/kafka/EmailSendMessageProducerIntegrationTest.java

```diff
@@ -104,16 +105,77 @@ void testSendOrderReceivedMessageToKafkaTopic() throws Exception {
+
+        confirmation.setForename("Jenny");
+        confirmation.setSurname("Wilson");
+
+        confirmation.setAddressLine1("Kemp House Capital Office");
+        confirmation.setAddressLine2("LTD");
+        confirmation.setHouseName("Kemp House");
+        confirmation.setHouseNumberStreetName("152-160 City Road");
+        confirmation.setCity("London");
+        confirmation.setPostCode("EC1V 2NX");
+        confirmation.setOrderReferenceNumber(ORDER_REFERENCE);
+        confirmation.setEmailAddress("mail@globaloffshore.com");
+        confirmation.setTimeOfPayment(TIME_OF_PAYMENT_FORMATTER.format(LocalDateTime.now()));
+        confirmation.setPaymentReference("RS5VSNDRE");
+        confirmation.setCompanyName("GLOBAL OFFSHORE HOST LIMITED");
+        confirmation.setCompanyNumber("11260147");
+        final ItemDetails missingImage = new ItemDetails();
+        missingImage.setDateFiled("15 Feb 2018");
+        missingImage.setDescription("Appointment of Ms Sharon Michelle White as a director on 4 February 2020");
+        missingImage.setFee("3");
+        missingImage.setType("AP01");
+        confirmation.setItemDetails(singletonList(missingImage));
+
+        final EmailSend email = new EmailSend();
+        email.setAppId("item-handler.missing-image-delivery-order-confirmation");
+        email.setEmailAddress("test@test.com");
+        email.setMessageId(UUID.randomUUID().toString());
+        email.setMessageType("missing_image_delivery_order_confirmation_email");
+        email.setData(objectMapper.writeValueAsString(confirmation));
+        email.setCreatedAt(LocalDateTime.now().toString());
+
+        // When email-send message is sent to kafka topic
+        final List<Message> messages = sendAndConsumeMessage(email);
+
+        // Then we can successfully consume the message.
+        assertThat(messages.isEmpty(), is(false));
+        byte[] consumedMessageSerialized = messages.get(0).getValue();
+        final String deserializedConsumedMessage = new String(consumedMessageSerialized);
+
+        // and it matches the serialized email-send object sent
+        final AvroSerializer<EmailSend> serializer = serializerFactory.getGenericRecordSerializer(EmailSend.class);
+        final byte[] serializedEmail = serializer.toBinary(email);
+        final String deserializedEmail = new String(serializedEmail);
+
+        assertEquals(deserializedConsumedMessage, deserializedEmail);
+    }
+
     private List<Message> sendAndConsumeMessage(final EmailSend email) throws Exception {
         List<Message> messages;
         testEmailSendMessageConsumer.connect();
         int count = 0;
         do {
             messages = testEmailSendMessageConsumer.pollConsumerGroup();
-            emailSendMessageProducerUnderTest.sendMessage(email, ORDER_REFERENCE);
+            sendMessageOnce(email, count);
             count++;
         } while (messages.isEmpty() && count < 15);
 
         return messages;
```

Comment:
What is the purpose of this method? It looks as originally it would keep sending messages, now it only sends it once?

Reply:
I found in some of the tests I ran that two messages were getting sent. That wasn't a (visible) issue when we only had one test in this test class, but once we had two, whichever test ran second would often fail because the second message from the first test would be picked up, instead of the (first) one sent for the second test.

Reply:
I see, I'll approve. But is there a way to reset the messages before each test?

Reply:
Thanks Nabeel. Good point. I'll look into that.

Reply:
My impression is that the right (simple) way to do this would be to send the message once before going into this polling loop. However, it seems that for that to work properly you need to set the auto offset reset behaviour on the consumer to earliest rather than the latest which is the default. This is so that it picks up messages which are already there. Unfortunately, as this consumer is using a `CHKafkaConsumerGroup`, we don't have control over that behaviour. We could at some point investigate tweaking the CH library or replacing the consumer in this test with a Spring Kafka consumer for example, but I think this should be looked into right now?

Reply:
Hey Vaughan, It's up to you. But the above works and I'm happy with that.

Reply:
Yes, I think we can live with this test as it is, as it does work, and the method was named to make it explicit that we want to limit the number of messages produced to 1. 👍

Reply:
Noted this in low-priority bug https://companieshouse.atlassian.net/browse/GCI-1519.